### DISCUSSION TITLE
我在登录页面调了登录请求，用localStorage存了数据，为什么在getInitialState无法获取到

### DISCUSSION THREAD

**Discussion by liushunli** [umijs/umi#11713] [Q&A] [answered]
![image](https://github.com/umijs/umi/assets/12014406/d1ca1946-0309-4740-a632-64fd2a2e1063)

![image](https://github.com/umijs/umi/assets/12014406/e159922a-04b4-432b-addf-a54ec5e2bc5e)

app.tsx获取不到
![image](https://github.com/umijs/umi/assets/12014406/c92e75c8-71c6-492f-ae9b-15262e2d679e)

没有昵称
![image](https://github.com/umijs/umi/assets/12014406/233c888d-4a6e-46b9-ad0b-c560c798b751)

**Comment by liushunli** [comment]
我怎么把登录存在localStorge的值弄到头像旁边

**Comment by fz6m** [comment, marked as answer]
理论上要判断用户有没有权限，一般通过 token 或 cookie 的鉴权方式来请求后端 api 判断，如果有权限，就继续请求 api 获取该用户的信息，所以无需把用户个人信息存在本地，应该 api 化。

如果有特殊原因要存在本地 local storage 中一些缓存信息，应该在登录成功时存在 local storage 中，之后自动刷新页面，此时就可以在 `getInitialState` 获取到了，因为 `getInitialState` 是整个应用的全局状态获取时机，在应用启动时就最先运行了。

正确的数据流应该是在 `getInitialState` 或者全局状态管理中请求后端 api 获取到该用户的个人信息并保存到 全局状态管理 或 initialState 的 model 中，之后在代码的任意地方使用。
